service/raft/raft_group0.cc

```diff
@@ -294,15 +300,22 @@ future<> raft_group0::join_group0(std::vector<raft::server_info> seeds, bool as_
             server = grp.server.get();
             co_await grp.persistence.bootstrap(std::move(initial_configuration));
             co_await _raft_gr.start_server_for_group(std::move(grp));
+            // FIXME if we crash now or after getting added to the config but before storing group 0 ID,
+            // we'll end with a bootstrapped server that possibly added some entries, but we won't remember that we have such a server
+            // after we restart. Then we'll call `persistence.bootstrap` again after restart which will overwrite our snapshot, leading to
+            // possibly incorrect state. One way of handling this may be changing `persistence.bootstrap` so it checks if any persistent
+            // state is present, and if it is, do nothing.
         }
+
+        assert(server);
         if (server->get_configuration().contains(my_id)) {
-            // True if we started a new group or completed a configuration change
-            // initiated earlier.
-            rslog.trace("server {} already in group as {}", my_id,
+            // True if we started a new group or completed a configuration change initiated earlier.
+            rslog.info("server {} already in group 0 (id {}) as {}", group0_id, my_id,
                     server->get_configuration().can_vote(my_id)? "voter" : "non-voter");
             break;
         }
-        auto pause_shutdown = _shutdown_gate.hold();
+
+        // XXX is 1 second timeout enough in a multi-dc cluster?
```

Comment:
I mean, seriously, how many of these XXX is one going to have to validate? They are going to stay forever even if they are total nonsense! And yes 1 second is enough for everybody.